Dashboard/app/js/lib/views/surveys/form-share.jsx

```diff
@@ -0,0 +1,95 @@
+/* eslint-disable import/no-unresolved */
+import React from 'react';
+import observe from 'akvo-flow/mixins/observe';
+import WebFormShare from 'akvo-flow/components/forms/form-share';
+
+require('akvo-flow/views/react-component');
+
+FLOW.WebFormShareView = FLOW.ReactComponentView.extend(
+  observe({
+    'FLOW.selectedControl.selectedSurvey.status': 'formValidation',
+    'FLOW.selectedControl.selectedSurveyGroup.monitoringGroup': 'formValidation',
+    'FLOW.questionControl.content.isLoaded': 'formValidation',
+    'FLOW.questionGroupControl.content.isLoaded': 'formValidation',
+    'FLOW.surveyControl.webformId': 'renderReactSide',
+  }),
+  {
+    init() {
+      this._super();
+      this.formValidation = this.formValidation.bind(this);
+      this.renderReactSide = this.renderReactSide.bind(this);
+      this.getProps = this.getProps.bind(this);
+      this.getShareURL = this.getShareURL.bind(this);
+
+      this.valid = false;
+    },
+
+    didInsertElement(...args) {
+      this._super(...args);
+
+      // call validate on init
+      this.formValidation();
+
+      this.renderReactSide();
+    },
+
+    // react side
+    renderReactSide() {
+      const props = this.getProps();
+      this.reactRender(<WebFormShare {...props} />);
+    },
+
+    getProps() {
+      return {
+        strings: {},
+        data: {
+          valid: this.valid,
+          shareUrl:
+            FLOW.surveyControl.webformId &&
+            `${window.location.origin}/webforms/${FLOW.surveyControl.webformId}`,
+        },
+        actions: {
+          getShareURL: this.getShareURL,
+        },
+      };
+    },
+
+    formValidation() {
+      const selectedForm = FLOW.selectedControl.get('selectedSurvey');
+      const selectedSurveyGroup = FLOW.selectedControl.get('selectedSurveyGroup');
+      const questions = FLOW.questionControl.get('content');
+      const questionGroups = FLOW.questionGroupControl.get('content');
+
+      // case 1 is form published?
+      const isPublished = selectedForm.get('status') === 'PUBLISHED';
+
+      // case 2 is not monitoring
+      const isNotMonitoring = selectedSurveyGroup.get('monitoringGroup') === false;
+
+      // case 3 does not have illegal question type
+      const noIllegalQuestion =
+        questions.some(question => {
+          switch (question.get('type')) {
+            case 'CADDISFLY':
+              return true;
+            case 'SIGNATURE':
+              return true;
+            case 'GEOSHAPE':
```

Comment:
backend side also checks for CASCADE type https://github.com/akvo/akvo-flow/pull/3564/files#diff-0b87b53604405485c8652761c8c3f11eR28 ... is that correct? @muloem

Reply:
True. That one slipped by me! 🤦  Cascade is supported! 
https://github.com/akvo/akvo-flow/issues/3145#issuecomment-630113567

Reply:
Huh, supported or not supported?

Reply:
Webforms can render cascade questions correctly so no need to disable sharing if there is a cascade question.